.ci/openshift-ci/build-root/Dockerfile

```diff
@@ -0,0 +1,78 @@
+# Use the centos image as the base image
+FROM quay.io/centos/centos:stream8
+
+ENV VERSION=1.19 \
+    GOCACHE=/go/.cache \
+    GOARM=5 \
+    GOPATH=/go \
+    GOROOT=/usr/local/go \
+    LOGNAME=deadbeef
+ENV PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+
+# Do all the stuff related with Golang, see https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/ci-tools-build-root.yaml#L338
+RUN yum install -y epel-release && \
+    rpm -V epel-release && \
+    INSTALL_PKGS="bc bind-utils bsdtar device-mapper device-mapper-persistent-data e2fsprogs ethtool file findutils gcc git glib2-devel hostname iptables jq krb5-devel libseccomp-devel libvirt-devel lsof make mercurial nmap-ncat openssl rsync socat systemd-devel tar tito tree util-linux wget which xfsprogs zip" && \
+    yum install -y $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum clean all && \
+    touch /os-build-image && \
+    git config --system user.name fabiobrz && \
+    git config --system user.email fburzigo@redhat.com
+
+RUN \
+    set -x &&  \
+    curl --fail --retry 3 -L https://golang.org/dl/go$VERSION.linux-amd64.tar.gz | tar -C /usr/local -xzf - && \
+    go install golang.org/x/tools/cmd/cover@latest && \
+    go install golang.org/x/tools/cmd/goimports@latest && \
+    go install golang.org/x/lint/golint@latest && \
+    go install gotest.tools/gotestsum@v1.9.0 && \
+    mv $GOPATH/bin/* /usr/bin/ && \
+    rm -rf $GOPATH/* $GOPATH/.cache && \
+    mkdir $GOPATH/bin && \
+    ln -s /usr/bin/goimports $GOPATH/bin/goimports
+
+RUN chmod g+xw -R $GOPATH && \
+    chmod g+xw -R $(go env GOROOT)
+
+# Downloading and installing Java
+ENV JAVA_PKG=https://download.oracle.com/java/17/latest/jdk-17_linux-x64_bin.tar.gz \
+      JAVA_HOME=/usr/java/jdk-17
+
+RUN set -eux; \
+      JAVA_SHA256=$(curl "$JAVA_PKG".sha256) ; \
+      curl --output /tmp/jdk.tgz "$JAVA_PKG" && \
+      echo "$JAVA_SHA256 */tmp/jdk.tgz" | sha256sum -c; \
+      mkdir -p "$JAVA_HOME"; \
+      tar --extract --file /tmp/jdk.tgz --directory "$JAVA_HOME" --strip-components 1
```

Comment:
I think we can simplify this a lot. 

- use more recent `dnf` instead of `yum`
- we need probably need only `git` (I'm not sure what native tools use Intersmash tests)
- OpenJDK can be installed from RPM as well

Reply:
> we need probably need only git (I'm not sure what native tools use Intersmash tests)

Maybe here you're relating to the following line?

`INSTALL_PKGS="bc bind-utils bsdtar device-mapper device-mapper-persistent-data e2fsprogs ethtool file findutils gcc git glib2-devel hostname iptables jq krb5-devel libseccomp-devel libvirt-devel lsof make mercurial nmap-ncat openssl rsync socat systemd-devel tar tito tree util-linux wget which xfsprogs zip"`

In suche a case, I just copy pasted from https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/ci-tools-build-root.yaml#L338

So are you suggesting to remove all this stuff?

Reply:
yes, exactly, we can probably install only `git` and `httpd-tools`

> Golang is needed per OpenShift CI docs requirements, see https://docs.ci.openshift.org/docs/architecture/ci-operator/#build-root-image

Ok, but I'm not really convinced it is really needed or used. 

> 1. What would be the value add for that? We'd got stuck with versioning in repositories. Why should we deal with such a pain for a builder image?

What pain do you mean? We would install the latest OpenJDK 17 from RPM instead of the latest Oracle JDK 17. 

> JDK and Maven are needed, since Intersmash is a Java/Maven project
> oc is needed in order to create users and perform other config steps

I don't suggest removing anything of these, only changing the way how JDK is installed. (I might have selected wrong lines in review)

Reply:
I don't know how repositories work in CentOS, but with RHEL (with subscription)/Fedore we could also drop `epel-release` repository

Reply:
+1 let's see whether it works. Golang is gone too.

Reply:
It works, thanks, the images are definitely smaller now, and the overall time for the test improved a lot, see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/37965/rehearse-37965-pull-ci-Intersmash-intersmash-openshift-ci.exp-intersmash-e2e/1643607318870364160